CorsixTH/Lua/graphics.lua

```diff
@@ -199,8 +199,8 @@ function Graphics:makeGreyscaleGhost(pal)
     local g_index = 0
     local g_diff = 100000 -- greater than 3*63^2 (TH uses 6 bit colour channels)
     for j = 0, #entries do
-      local entry = entries[j]
-      local diff = (entry[1] - g)^2 + (entry[2] - g)^2  + (entry[3] - g)^2
+      local entry_j = entries[j]
```

Comment:
There is a lot more wrong here (and elsewhere too), my primary aim is to fix all errors reported by luacheck rather than really improving the local code.

EDIT: I rewrote it, but the longer I look at it, the less sense it makes. The magic numbers to convert to grey scale look like the regular conversion values, but corsix uses 6 bit colour values, as it says in the comment, so shouldn't that get compensated somewhere in the greyscale conversion, and then back to the 6 bit value in the diff calculation?